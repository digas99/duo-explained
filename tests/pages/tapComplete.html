<!DOCTYPE html>
<html data-duo-theme="dark" dir="ltr" lang="en"><head><title data-react-helmet="true">Duolingo - The world's best way to learn French</title> </head><body style="--onetrust-banner-height: 0px;"><div class="fs-unmask" id="root" onclick=""><div class="kPqwA _2kkzG"><div class="_3yE3H"><div class="_2G9ED _29gfw"><div class="wqSzE"><div class="_3v4ux"><div class="I-Avc _1zcW8"><button class="_3vGNs _2YF0P _3X5b9" data-test="quit-button"><img class="_3X5b9 _1o-YO" src="./tapComplete_files/df223d5b9feb8017b323ed21103eb5ac.svg"/></button><div aria-valuemax="1" aria-valuemin="0" aria-valuenow="0.18266030526357188" class="_358Nq" role="progressbar" style="--web-ui_progress-bar-color: rgb(var(--color-owl)); --web-ui_progress-bar-shine-height: 3px; --__internal__progress-bar-height: 16px; --__internal__progress-bar-inner-value: 0%; --__internal__progress-bar-value: 18.27%;"><div class="_2-LJi"><div class="_2fuvy" style="opacity: 1;"><div class="_3Dzp2 _2fuvy"></div><div class="_1EzVJ"></div></div><div class="_3FEy7" style="opacity: 1;"><div class="_2fYtx"><span style="--progress-bar-height: 16px;"><span class="RxQcJ fs-exclude SVDhH"></span></span></div></div></div></div><div class="_3ww7z"><img src="./tapComplete_files/0c1a523b4b5882a97e4df162f4b5c58b.svg" style="height: 32px; width: 32px;"/><img src="./tapComplete_files/8d4c5fcd9cedabfd155d3eda8af269bc.svg"/></div></div></div></div><div class="RMEuZ _1GVfY"><div class="_3GuWo _1cTBC"><div class="_1XNQX _3V6my"><div class="_1fxa4 _1Mopf" data-test="challenge challenge-tapComplete"><div class="_2n5fx _1JTA4 _3B8G- K3vbJ"><div class="U41Ye"><h1 class="_3EOK0" data-test="challenge-header">Fill in the blanks</h1></div><div class="_3Mge5 f7WE2 _3rat3"><div class="_35PZ2 _35mGI" dir="ltr"><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">C</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">'</span><div aria-label="C'" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 14.9219px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">e</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">s</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">t</span><div aria-label="est" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 25.0938px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">l</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">a</span><div aria-label="la" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 15.5781px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">p</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">r</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">e</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">m</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">i</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">è</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">r</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">e</span><div aria-label="première" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 75.0312px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">r</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">u</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">e</span><div aria-label="rue" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 27.7188px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><div class="_17wB6"><span class="_3AISd"><div class="_8g3cL"><span class="_3BS5Z _2tbfR" style="visibility: hidden;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q WxjqG _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i" data-test="au-challenge-tap-token" dir="ltr" lang="fr" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">au</span></span></button></span></span><span class="_3BS5Z"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q WxjqG _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i" data-test="à-challenge-tap-token" dir="ltr" lang="fr" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">à</span></span></button></span></span></div></span></div><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">d</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">r</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">o</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">i</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">t</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">e</span><div aria-label="droite" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 47.8594px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">.</span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">N</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">o</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">u</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">s</span><div aria-label="Nous" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 41.8125px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">a</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">l</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">l</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">o</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">n</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">s</span><div aria-label="allons" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 50.7188px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><div class="_17wB6"><span class="_3AISd"><div class="_8g3cL"><span class="_3BS5Z"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q WxjqG _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i" data-test="au-challenge-tap-token" dir="ltr" lang="fr" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">au</span></span></button></span></span><span class="_3BS5Z _2tbfR" style="visibility: hidden;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q WxjqG _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i" data-test="à-challenge-tap-token" dir="ltr" lang="fr" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">à</span></span></button></span></span></div></span></div><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">z</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">o</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">o</span><div aria-label="zoo" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 28.2344px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">.</span></span></div><div class="_1ywbs"><div class="eSgkc" data-test="word-bank"><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _31wR2" data-test="au-challenge-tap-token" dir="ltr" lang="fr" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">au</span></span></button></span></div><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _31wR2" data-test="à-challenge-tap-token" dir="ltr" lang="fr" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">à</span></span></button></span></div></div></div></div></div></div></div></div></div><div class="_3GuWo _1cTBC _1QQhE"><div class="_1XNQX"><div class="_3FB5S _1VTif _3yMvO _2HXQ9 d-cgpt-footer" id="session/PlayerFooter"><div class="U8jH3 jHbiF"><div class="AXBVw _2Ycbe _8wIx-"><div class="_1k6eg" data-test="blame blame-correct"><div class="_31ixh HJdJI fwpkb"><img alt="" class="_3oCTd" src="./tapComplete_files/b377ec812acb8c96d87d52e8009478ad.svg"/></div><div class="d6Xhl"><div class="_3LKiy"><div class="_1D3fo"><h2 class="_2U7Gm WXwlk">Nice!</h2></div></div><div class="_2yBiG"><div class="_10y_e"><button class="_1pDd6 _2JTJI bafGS _2LoNU VzbUl _1saKQ _1AgKJ"><span class="_24Rh2"><span class="_2e3N3 _3jt-c"></span><span class="_1qfbO _2Rt1l">Report</span></span></button></div></div></div></div></div><div class="MYehf d-cgpt-explain-button-wrapper"><button aria-disabled="false" class="_1x5JY _1M9iF _36g4N _2YF0P _3DbUj _38g3s _2oGJR d-cgpt-explain-button d-cgpt-explain-button-enabled" data-hasclick="true" id="d-cgpt-explain-button"><span class="_1o-YO">Explain</span></button><button aria-disabled="false" class="_1x5JY _1M9iF _36g4N _2YF0P _3DbUj _38g3s _2oGJR" data-test="player-next"><span class="_1o-YO">Continue</span></button></div></div></div>
<div class="d-cgpt-speech-bubble" style="bottom: 130px;">
<div>
<textarea placeholder="If you want, add to the query to ChatGPT..."></textarea>
<div class="d-cgpt-speech-bubble-tip"></div>
</div>
</div>
</div></div></div></div></div><div class="Vm8CO _1Fnem" data-test="drawer-backdrop" style="pointer-events: none;"></div></div><div class="fs-unmask" id="overlays" onclick=""><div class="Vm8CO _2zxQ8" style="pointer-events: none;"></div><a class="_1Qh5D _36g4N _2YF0P _3dDzz _1wiIJ" href="https://preview.duolingo.com/help/support-request" rel="noopener" target="_blank" title="Report feedback"><img class="_3QUr-" src="./tapComplete_files/b23d90ca7ea01eb330158d4939dc4899.svg"/></a><div data-focus-guard="true" style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;" tabindex="-1"></div><div data-focus-lock-disabled="disabled"></div><div data-focus-guard="true" style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;" tabindex="-1"></div></div><div><div class="grecaptcha-badge" data-style="bottomright" style="width: 256px; height: 60px; display: block; transition: right 0.3s; position: fixed; bottom: 14px; right: -186px; box-shadow: gray 0px 0px 5px; border-radius: 2px; overflow: hidden;"><div class="grecaptcha-logo"></div><div class="grecaptcha-error"></div><textarea class="g-recaptcha-response" id="g-recaptcha-response-100000" name="g-recaptcha-response" style="width: 250px; height: 40px; border: 1px solid rgb(193, 193, 193); margin: 10px 25px; padding: 0px; resize: none; display: none;"></textarea></div></div><div id="onetrust-consent-sdk"><div class="onetrust-pc-dark-filter ot-hide ot-fade-in"></div></div></body></html>