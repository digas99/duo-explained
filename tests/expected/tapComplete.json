{
    "content": {
        "sentence": "C'est la première rue <blank> droite. Nous allons <blank> zoo.",
        "answer": "C'est la première rue à droite. Nous allons au zoo.",
        "wordBank": [
			"au",
            "à"
        ]
    },
    "wrapper": {},
    "exercise": "Fill in the blanks",
    "type": "tapComplete",
    "language": {
        "source": "fr",
        "target": "fr",
        "ui": {
			"client": {
				"main": "en-US",
				"options": ["en-US"]
			},
			"document": "en"
		}
    }
}